schemas/dataSources.properties.schema.json

```diff
@@ -0,0 +1,47 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema",
+    "id": "https://immersive-web/gamepad-mapping/0.1.0/dataSources.properties.schema.json",
+    "type": "object",
+    "additionalProperties": false,
+    "definitions": {
+        "id": {
+            "description" : "An identifier for the data source",
+            "type" : "string"
+        },
+        "button": {
+            "type": "object",
+            "additionalProperties": false,
+            "required": ["gamepadButtonsIndex"],
+            "properties": {
+                "gamepadButtonsIndex": { 
+                    "allOf": [{ "$ref": "gamepadIndex.schema.json" }]
+                },
+                "supportsTouch" : {
+                    "description" : "Indicates all of the buttons is capable of reporting GamepadButton.touched",
+                    "type" : "boolean",
+                    "default" : "false"
+                },
+                "supportsPress" : {
+                    "description" : "Indicates all of the buttons is capable of reporting GamepadButton.pressed",
+                    "type" : "boolean",
+                    "default" : "true"
+                },
+                "analogValues" : {
+                    "description" : "Indicates at least one of the buttons are capable of reporting numbers in between min and max",
+                    "type" : "boolean",
+                    "default" : "false"
+                }
+            }
+        },
+        "axis": {
+            "type": "object",
+            "additionalProperties": false,
+            "required": ["gamepadAxesIndex"],
+            "properties": {
+                "gamepadAxesIndex": { 
+                    "allOf": [{ "$ref": "gamepadIndex.schema.json" }]
+                }
+            }
+        }
+    }
```

Comment:
One thing that I would like to see added to this mapping is a way to flag an axis or button as "empty". The nature of the Gamepad API's `mapping` system is that if a controller lacks an input it should just leave a gap in the array. For example: The Daydream controller, since it lacks a trigger, would have a button[0] that never reports non-zero values under the current Gamepad API proposal. But there's no native way to differentiate between a button that's not present vs. one that's simply never pressed, so it's an area where the schema would have to step in and help.

Reply:
There isn't a way to differentiate that?  Wouldn't the buttons array element just be null?

Reply:
Hrm... I'm not sure that the element can be null. In order to allow that the Gamepad spec would have to indicate that it's a ` FrozenArray<GamepadButton?>`, but it currently specifies ` FrozenArray<GamepadButton>`. As far as I know there's no way that the gamepad API provides to indicate actual gaps in the button/axes mapping.

Reply:
Erk... well @thetuvix isn't going to be happy to hear that....